GMG/GMG/UI/Components/Metronome.swift

```diff
@@ -6,13 +6,308 @@
+            let player = try AVAudioPlayer(contentsOf: url)
+            
+            player.volume = 1.0
+            player.prepareToPlay()
+            
+            self.lowPlayer = player
+        } catch {
+            print("[Metronome] Failed to initialize audio player for low")
+        }
+    }
+    
+    private func unloadHighPlayer() {
+        guard let highPlayer else { return }
+        highPlayer.stop()
+        
+        self.highPlayer = nil
+    }
+    
+    private func unloadLowPlayer() {
+        guard let lowPlayer else { return }
+        lowPlayer.stop()
+        
+        self.lowPlayer = nil
+    }
+    
+    // 소리 중 강박을 구분을 위한 인텍스 설정 (3/4, 4/4 박자는 첫 번째 박만 강박이지만, 6/8 박자의 경우 1, 4번 박에 강박을 줘야 함) *강박: Metronome 재생 중 기준을 잡는 높은 음
+    private func shouldUseHigh(beatIndex: Int) -> Bool {
+        if timeSignature.numerator == 6 && timeSignature.denumerator == .eighth {
+            return beatIndex == 0 || beatIndex == 3
+        } else {
+            return beatIndex == 0
+        }
+    }
+    
+    // 박자별 High or Low 음성 재생
+    private func playBeatSound(for beatIndex: Int) {
+        if shouldUseHigh(beatIndex: beatIndex) {
+            guard let audioPlayer = highPlayer else {
+                print("[Metronome] highPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        } else {
+            guard let audioPlayer = lowPlayer else {
+                print("[Metronome] lowPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        }
     }
 }
 
 #Preview {
-    Metronome()
+    struct PreviewWrapper: View {
+        @State private var bpm: Int = 100
+        @State private var isPlaying: Bool = false
+        @State private var timeSignature: TimeSignature = .fourFour
+
+        var body: some View {
+            VStack(spacing: 20) {
+                Metronome(
+                    bpm: BPM(value: bpm),
+                    isPlaying: isPlaying,
+                    timeSignature: timeSignature
+                )
+
+                Button(isPlaying ? "Stop" : "Play") {
+                    isPlaying.toggle()
+                }
+                .buttonStyle(.bordered)
+
+                Stepper("BPM: \(bpm)", value: $bpm, in: 30...320)
+                
+                HStack(spacing: 30) {
+                    Button("3/4") {
+                        timeSignature = .threeFour
+                    }
+                    .buttonStyle(.bordered)
+                    Button("4/4") {
+                        timeSignature = .fourFour
+                    }
+                    .buttonStyle(.bordered)
+                    Button("6/8") {
+                        timeSignature = .sixEight
+                    }
+                    .buttonStyle(.bordered)
+                }
+            }
+            .padding()
+        }
+    }
+
+    return PreviewWrapper()
```

Comment:
`@Previewable` 쓰면 따로 뷰 안만들어도 돼요!

```suggestion
    @Previewable @State var bpm: Int = 100
    @Previewable @State var isPlaying: Bool = false
    @Previewable @State var timeSignature: TimeSignature = .fourFour

    VStack(spacing: 20) {
        Metronome(
            bpm: BPM(value: bpm),
            isPlaying: isPlaying,
            timeSignature: timeSignature
        )

        Button(isPlaying ? "Stop" : "Play") {
            isPlaying.toggle()
        }
        .buttonStyle(.bordered)

        Stepper("BPM: \(bpm)", value: $bpm, in: 30...320)

        HStack(spacing: 30) {
            Button("3/4") {
                timeSignature = .threeFour
            }
            .buttonStyle(.bordered)
            Button("4/4") {
                timeSignature = .fourFour
            }
            .buttonStyle(.bordered)
            Button("6/8") {
                timeSignature = .sixEight
            }
            .buttonStyle(.bordered)
        }
    }
    .padding()
```

Reply:
오오 이런 방법이 있었군요! 감사합니다 수정할게요!!